backend/modules/sse/sse_consumer.py

```diff
@@ -0,0 +1,134 @@
+import asyncio
+import logging
+from abc import ABC, abstractmethod
+from urllib.parse import parse_qs
+
+from channels.generic.http import AsyncHttpConsumer
+from typing_extensions import override
+
+from config.settings import CORS_ALLOWED_ORIGINS
+
+
+class SSEConsumer(AsyncHttpConsumer, ABC):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.sse_task = None
+        self.shutdown_event = asyncio.Event()
+
+    @staticmethod
+    def _headers(response_origin: bytes | None):
+        return [
+            (b"Content-Type", b"text/event-stream"),
+            (b"Cache-Control", b"no-cache"),
+            (b"Connection", b"keep-alive"),
+            (b"Access-Control-Allow-Origin", response_origin or b"*"),
+            (b"Access-Control-Allow-Methods", b"GET, OPTIONS"),
+            (b"Access-Control-Allow-Headers", b"Content-Type, Authorization"),
+        ]
+
+    async def handle_preflight(self, origin):
+        origin_str = origin.decode("utf-8") if origin else ""
+        response_origin = origin if origin_str in CORS_ALLOWED_ORIGINS else b""
+        await self.send_response(
+            status=204,
+            headers=self._headers(response_origin),
+            body=b"",
+        )
+        return None
+
+    @staticmethod
+    @abstractmethod
+    async def _validate_auth(bearer_token: str) -> bool:
+        """
+        Validates the provided bearer token using Clerk authentication.
+        :param bearer_token: The token to validate.
+        :return: bool: True if the token is valid, False otherwise.
+        """
+        raise NotImplementedError("Subclasses must implement validate_auth method.")
+
+    async def http_request(self, message) -> None:
+        """
+        Async entrypoint for the HTTP request.
+        This method now sets up the SSE connection and starts a background
+        task to send events, rather than blocking.
+        """
+        headers: dict[bytes, bytes] = dict(self.scope["headers"])
+        request_origin = headers.get(b"origin", b"")
+
+        if self.scope["method"] == "OPTIONS":
+            return await self.handle_preflight(request_origin)
+
+        auth_header = headers.get(b"authorization", b"")
+        if auth_header.startswith(b"Bearer "):
+            bearer_token = auth_header[7:]
```

Comment:
i feel that doing split would be more readable here
```suggestion
            bearer_token = auth_header.split()[1]
```